core/steps/cloudinit/disk_mount.go

```diff
@@ -0,0 +1,142 @@
+package cloudinit
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+
+	"github.com/weaveworks-liquidmetal/flintlock/client/cloudinit"
+	"github.com/weaveworks-liquidmetal/flintlock/client/cloudinit/userdata"
+	"github.com/weaveworks-liquidmetal/flintlock/core/models"
+	"github.com/weaveworks-liquidmetal/flintlock/pkg/log"
+	"github.com/weaveworks-liquidmetal/flintlock/pkg/planner"
+)
+
+func NewDiskMountStep(vm *models.MicroVM) planner.Procedure {
+	return &diskMountStep{
+		vm: vm,
+	}
+}
+
+type diskMountStep struct {
+	vm *models.MicroVM
+}
+
+// Name is the name of the procedure/operation.
+func (s *diskMountStep) Name() string {
+	return "cloudinit_disk_mount"
+}
+
+func (s *diskMountStep) ShouldDo(ctx context.Context) (bool, error) {
+	logger := log.GetLogger(ctx).WithFields(logrus.Fields{
+		"step": s.Name(),
+	})
+	logger.Debug("checking if procedure should be run")
+
+	if !s.vm.Spec.AdditionalVolumes.HasMountableVolumes() {
+		return false, nil
+	}
+
+	for _, vol := range s.vm.Spec.AdditionalVolumes {
+		if vol.MountPoint == "" {
+			continue
+		}
+
+		status := s.vm.Status.Volumes[vol.ID]
+
+		if status == nil || status.Mount.Source == "" {
+			return true, nil
+		}
+	}
+
+	vendorData, err := s.getVendorData()
+	if err != nil {
+		return false, fmt.Errorf("getting vendor data: %w", err)
+	}
+	if vendorData == nil {
+		return true, nil
+	}
+
+	for _, vol := range s.vm.Spec.AdditionalVolumes {
+		if vol.MountPoint == "" {
+			continue
+		}
+
+		if !vendorData.HasMountByMountPoint(vol.MountPoint) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// Do will perform the operation/procedure.
+func (s *diskMountStep) Do(ctx context.Context) ([]planner.Procedure, error) {
+	logger := log.GetLogger(ctx).WithFields(logrus.Fields{
+		"step": s.Name(),
+	})
+	logger.Debug("running step to mount additional disks via cloud-init")
+
+	vendorData, err := s.getVendorData()
+	if err != nil {
+		return nil, fmt.Errorf("getting vendor data: %w", err)
+	}
+	if vendorData == nil {
+		vendorData = &userdata.UserData{}
+	}
+
+	startingCode := int('b')
+	for i, vol := range s.vm.Spec.AdditionalVolumes {
+		if vol.MountPoint == "" {
+			continue
+		}
+
+		device := fmt.Sprintf("vd%c", rune(startingCode+i)) // Device number is always +1 as we have the root volume first
```

Comment:
IIRC it is network interface names it ignores, it will always `eth0`, `eth1`, etc in order no matter what you ask for. Would be good to check whether it does something similar for volumes tho.

Reply:
It was the network interface names.